sagemaker-python-sdk/tensorflow_distributed_mnist/tensorflow_batch_transform_mnist.ipynb

```diff
@@ -266,22 +234,87 @@
     "import json\n",
     "from urllib.parse import urlparse\n",
     "\n",
+    "import boto3\n",
+    "\n",
     "parsed_url = urlparse(transformer.output_path)\n",
     "bucket_name = parsed_url.netloc\n",
-    "file_key = '{}/{}.out'.format(parsed_url.path[1:], input_file_name)\n",
+    "prefix = parsed_url.path[1:]\n",
     "\n",
-    "output_obj = s3.Object(bucket_name, file_key)\n",
-    "output = output_obj.get()[\"Body\"].read()\n",
+    "s3 = boto3.resource('s3')\n",
     "\n",
-    "print('Prediction is {}'.format(json.loads(output)['outputs']['classes']['int64Val']))"
+    "predictions = []\n",
+    "for i in range(10):\n",
+    "    file_key = '{}/data-{}.csv.out'.format(prefix, i)\n",
+    "\n",
+    "    output_obj = s3.Object(bucket_name, file_key)\n",
+    "    output = output_obj.get()[\"Body\"].read().decode('utf-8')\n",
+    "\n",
+    "    predictions.extend(json.loads(output)['outputs']['classes']['int64Val'])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For demonstration purposes, we're also going to download the original input data so that we can see how the model did with its predictions."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "\n",
+    "import matplotlib.pyplot as plt\n",
+    "from numpy import genfromtxt\n",
+    "\n",
+    "plt.rcParams['figure.figsize'] = (2,10)\n",
+    "\n",
+    "def show_digit(img, caption='', subplot=None):\n",
+    "    if subplot == None:\n",
+    "        _,(subplot) = plt.subplots(1,1)\n",
+    "    imgr = img.reshape((28,28))\n",
+    "    subplot.axis('off')\n",
+    "    subplot.imshow(imgr, cmap='gray')\n",
+    "    plt.title(caption)\n",
+    "\n",
+    "tmp_dir = '/tmp/data'\n",
+    "if not os.path.exists(tmp_dir):\n",
+    "    os.makedirs(tmp_dir)\n",
+    "\n",
+    "for i in range(10):\n",
+    "    input_file_name = 'data-{}.csv'.format(i)\n",
+    "    input_file_key = '{}/{}'.format(input_file_path, input_file_name)\n",
+    "    \n",
+    "    s3.Bucket(input_bucket_name).download_file(input_file_key, os.path.join(tmp_dir, input_file_name))\n",
```

Comment:
Get a `ClientError: An error occurred (404) when calling the HeadObject operation: Not Found` error.

Reply:
I still got an error here.  It looks like `input_file_path` still has a trailing "/", which means `input_file_key` has a double slash.  If you change it to:

```
    input_file_key = '{}{}'.format(input_file_path, input_file_name)
```

everything works as expected.

Reply:
fixed - removed the trailing slash above